### PR TITLE
Add VK_PIPELINE_STAGE_CONDITIONAL_RENDERING_BIT_EXT in supported_pipeline_stages_table

### PR DESCRIPTION
**Pull request by locke-lunarg**
https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/507
@scyganINTEL

### CODE REVIEW

File: tests/vkpositivelayertests.cpp [hunk 1]
```diff
@@ -8626,3 +8626,81 @@ TEST_F(VkPositiveLayerTest, SwapchainImageLayout) {
     vk::DestroyFence(m_device->device(), fence, NULL);
     vk::DestroyImageView(m_device->device(), view, NULL);
 }
+
+TEST_F(VkPositiveLayerTest, PipelineStageConditionalRednering) {
```
Comment (Tony-LunarG) [inline tests/vkpositivelayertests.cpp]: Rednering -> Rendering

File: tests/vkpositivelayertests.cpp [hunk 2]
```diff
@@ -8626,3 +8626,81 @@ TEST_F(VkPositiveLayerTest, SwapchainImageLayout) {
...
+    imb.subresourceRange.aspectMask = VK_IMAGE_ASPECT_COLOR_BIT;
+    imb.subresourceRange.baseMipLevel = 0;
+    imb.subresourceRange.levelCount = 1;
+    imb.subresourceRange.baseArrayLayer = 0;
+    imb.subresourceRange.layerCount = 1;
+
+    m_errorMonitor->ExpectSuccess();
+    vk::CmdPipelineBarrier(m_commandBuffer->handle(), VK_PIPELINE_STAGE_VERTEX_SHADER_BIT,
+                           VK_PIPELINE_STAGE_CONDITIONAL_RENDERING_BIT_EXT, 0, 0, nullptr, 0, nullptr, 1, &imb);
+    m_errorMonitor->VerifyNotFound();
```
Comment (Tony-LunarG) [inline tests/vkpositivelayertests.cpp]: When I run the test, it passes with and without the validation layer changes,  Should it?

File: tests/vkpositivelayertests.cpp (hunk 1 above)
Comment (locke-lunarg) [reply tests/vkpositivelayertests.cpp]: fixed

File: tests/vkpositivelayertests.cpp (hunk 2 above)
Comment (locke-lunarg) [reply tests/vkpositivelayertests.cpp]: I added a test with wrong queue flags

File: tests/vklayertests_pipeline_shader.cpp [hunk 3]
```diff
@@ -6809,3 +6809,101 @@ TEST_F(VkLayerTest, RayTracingPipelineShaderGroups) {
         m_errorMonitor->VerifyFound();
     }
 }
+
+TEST_F(VkLayerTest, PipelineStageConditionalRenderingWithWrongQueue) {
+    TEST_DESCRIPTION("Run CmdPipelineBarrier with VK_PIPELINE_STAGE_CONDITIONAL_RENDERING_BIT_EXT and wrong VkQueueFlagBits");
+    ASSERT_NO_FATAL_FAILURE(Init());
+    // m_device->m_queue = m_device->dma_queues()[0]->handle();
```
Comment (Tony-LunarG) [inline tests/vklayertests_pipeline_shader.cpp]: Leftover from debugging?

File: tests/vklayertests_pipeline_shader.cpp [hunk 4]
```diff
@@ -6809,3 +6809,101 @@ TEST_F(VkLayerTest, RayTracingPipelineShaderGroups) {
...
+    uint32_t only_transfer_queueFamilyIndex = UINT32_MAX;
+
+    const auto q_props = vk_testing::PhysicalDevice(gpu()).queue_properties();
+    ASSERT_TRUE(q_props.size() > 0);
+    ASSERT_TRUE(q_props[0].queueCount > 0);
+
+    for (uint32_t i = 0; i < (uint32_t)q_props.size(); i++) {
+        if (q_props[i].queueFlags == VK_QUEUE_TRANSFER_BIT) {
```
Comment (Tony-LunarG) [inline tests/vklayertests_pipeline_shader.cpp]: Not necessary, but if this test would tolerate VK_QUEUE_SPARSE_BINDING_BIT, it might run on more systems

Comment (locke-lunarg) [reply tests/vklayertests_pipeline_shader.cpp]: VK_QUEUE_SPARSE_BINDING_BIT doesn't work in CmdPipelineBarrier.

**Pull request merged**